### PR TITLE
Feature eyedropper

### PR DESCRIPTION
**Pull request by fsciuti**
Add EyeDropper API feature.

### CODE REVIEW

**Review by NOtherDev — APPROVED** [review]
Looks good 🙂 Two minor spelling comments.
Were you able to run the setup locally successfully @fsciuti ?

File: src/data/features/eyeDropper.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import {Feature} from "../../utils/feature.js";
+
+export default new Feature({
+  id: 'eye-dropper',
+  name: 'EyeDropper',
```
[the remark this answers is not in the document]
Comment (fsciuti) [reply src/data/features/eyeDropper.js:5]: Hi @NOtherDev, the feature is called EyeDropper API (single word), as you can see on MDN Doc:
https://github.com/NOtherDev/whatwebcando/pull/92#pullrequestreview-883442042

File: src/data/features/eyeDropper.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  description: [
+    `The <b>EyeDropper API</b> allows users to catch sample colors from their screen with an eyedropper tool.`,
+    `Unlike <code>&lt;input type="color"&gt;</code>  on Chromium-based Desktop browsers, this API offers a simple interface to pick the colour of the entire device screen with a standard API.`,
```
[the remark this answers is not in the document]
Comment (fsciuti) [reply src/data/features/eyeDropper.js:8]: Yes! You are right!

File: src/data/features/eyeDropper.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (NOtherDev) [reply src/data/features/eyeDropper.js:5]: Yes, the API might be called this way. But I always wanted to describe features (business/user-oriented thing), not APIs (tech-oriented thing). I think, for the user, the feature would be called "Eye Dropper", and it is handled by the API called "EyeDropper API" and that's ok. Like "Camera Access" feature is done via "getUserMedia API" - the user-facing and the technical name might not necessarily match. Hope you see my point.

**Comment by fsciuti** [comment]
> Looks good 🙂 Two minor spelling comments. Were you able to run the setup locally successfully @fsciuti ?

Yes...I used nvm to run Node lts/erbium, otherwise Sapper can't run correctly.

File: src/data/features/eyeDropper.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (fsciuti) [reply src/data/features/eyeDropper.js:5]: Yeah! I understand and I share your point of view.

**Comment by NOtherDev** [comment]
Thank you @fsciuti ! 
I double checked the deploy instruction - it's `npm run export` and then `npm run deploy`. Go on and try it yourself - good luck! 💪

**Comment by fsciuti** [comment]
I tried but I received a lot of errors... I'll check soon and let you know!

Francesco Sciuti

CEO @ Devmy - Software Factory & Learning Google Developer Expert - Web
Technologies, & Microsoft MVP


[image: Google Developer Expert]
<https://developers.google.com/community/experts/directory/profile/profile-francesco-sciuti>[image:
Microsoft MVP]
<https://mvp.microsoft.com/it-it/PublicProfile/5003974?fullName=Francesco+Sciuti>
<https://www.meetup.com/it-IT/AWS-User-Group-Catania/>


Tel./Whatsapp: +393920445357

*Web: https:// <https://devmy.it/>**devmy.it <https://devmy.it/>*




*Facebook: https://www.facebook.com/Devmy.it
<https://www.facebook.com/Devmy.it>Linkedin:
https://www.linkedin.com/company/devmyfactory/
<https://www.linkedin.com/company/devmyfactory/>Twitter:
https://twitter.com/DevmyFactory
<https://twitter.com/DevmyFactory>Instagram:
https://www.instagram.com/Devmy.it/
<https://www.instagram.com/Devmy.it/>*

*Youtube: https://www.youtube.com/c/devmy <https://www.youtube.com/c/devmy>*



*Francesco è anche:**- Google Certified Certified - Mobile Web Specialist*

*- AWS User Group Leader*


*- Open JS Foundation Supporter*


.    <https://www.meetup.com/it-IT/AWS-User-Group-Catania/>
<https://www.credential.net/e7ba4674-48c3-4442-8af3-449fec881496>


Devmy è un marchio di Acadevmy s.r.l. - P.I. 05537900879

Questa e-mail contiene informazioni di carattere confidenziale rivolte
esclusivamente al destinatario sopra indicato. E' vietato l'uso, la
diffusione, distribuzione o riproduzione da parte di ogni altra persona.
Nel caso aveste ricevuto questa e-mail  per errore, siete pregati di
segnalarlo immediatamente al mittente e distruggere quanto ricevuto senza
farne copia. Qualsivoglia utilizzo non autorizzato del contenuto di questa
e-mail costituisce violazione dell'obbligo di non prendere cognizione della
corrispondenza tra altri soggetti, salvo più grave illecito, ed espone il
responsabile alle relative conseguenze (D. Lgs. 196/2003).

The information contained in this message and any attachments hereto are
intended only for the personal use of the designated recipient(s) named
above. This message is a professional communication, and as such, is
privileged and confidential. If the reader of this message is not the
intended recipient, you are hereby notified that you have received this
document in error, and that any review, dissemination, distribution, or
copying of this message is prohibited. Please notify us immediately by
telephone.


Il giorno mer 16 feb 2022 alle ore 07:26 Adam Bar ***@***.***>
ha scritto:

**Comment by NOtherDev** [comment]
Ok it was only me who did it so far. I guess we'll have to go through the setup for the first time. The `dist` directory need to be an independent (`--orphan`) checkout of the `gh-pages` branch.

**Comment by fsciuti** [comment]
Can you explain better? I don't want to disturb you!

Francesco Sciuti

CEO @ Devmy - Software Factory & Learning Google Developer Expert - Web
Technologies, & Microsoft MVP


[image: Google Developer Expert]
<https://developers.google.com/community/experts/directory/profile/profile-francesco-sciuti>[image:
Microsoft MVP]
<https://mvp.microsoft.com/it-it/PublicProfile/5003974?fullName=Francesco+Sciuti>
<https://www.meetup.com/it-IT/AWS-User-Group-Catania/>


Tel./Whatsapp: +393920445357

*Web: https:// <https://devmy.it/>**devmy.it <https://devmy.it/>*




*Facebook: https://www.facebook.com/Devmy.it
<https://www.facebook.com/Devmy.it>Linkedin:
https://www.linkedin.com/company/devmyfactory/
<https://www.linkedin.com/company/devmyfactory/>Twitter:
https://twitter.com/DevmyFactory
<https://twitter.com/DevmyFactory>Instagram:
https://www.instagram.com/Devmy.it/
<https://www.instagram.com/Devmy.it/>*

*Youtube: https://www.youtube.com/c/devmy <https://www.youtube.com/c/devmy>*



*Francesco è anche:**- Google Certified Certified - Mobile Web Specialist*

*- AWS User Group Leader*


*- Open JS Foundation Supporter*


.    <https://www.meetup.com/it-IT/AWS-User-Group-Catania/>
<https://www.credential.net/e7ba4674-48c3-4442-8af3-449fec881496>


Devmy è un marchio di Acadevmy s.r.l. - P.I. 05537900879

Questa e-mail contiene informazioni di carattere confidenziale rivolte
esclusivamente al destinatario sopra indicato. E' vietato l'uso, la
diffusione, distribuzione o riproduzione da parte di ogni altra persona.
Nel caso aveste ricevuto questa e-mail  per errore, siete pregati di
segnalarlo immediatamente al mittente e distruggere quanto ricevuto senza
farne copia. Qualsivoglia utilizzo non autorizzato del contenuto di questa
e-mail costituisce violazione dell'obbligo di non prendere cognizione della
corrispondenza tra altri soggetti, salvo più grave illecito, ed espone il
responsabile alle relative conseguenze (D. Lgs. 196/2003).

The information contained in this message and any attachments hereto are
intended only for the personal use of the designated recipient(s) named
above. This message is a professional communication, and as such, is
privileged and confidential. If the reader of this message is not the
intended recipient, you are hereby notified that you have received this
document in error, and that any review, dissemination, distribution, or
copying of this message is prohibited. Please notify us immediately by
telephone.


Il giorno gio 17 feb 2022 alle ore 15:02 Adam Bar ***@***.***>
ha scritto:

**Comment by fsciuti** [comment]
Hi Adam, how are you?
I have researched your solution but what do you think if we use a solution
like one of these:

   -
   ***@***.***/deploying-sapper-pwa-using-github-pages-step-by-step-tutorial-part-1-1e9828dfe4de
   - https://dev.to/agustinl/deploy-sapper-pwa-on-github-pages-2ih1
   - https://gavinr.com/svelte-sapper-github-pages-actions/

I don't understand the reason for your choice on *npm run export *and *npm
run deploy *scripts.

Thank you very much and I apologize for the inconvenience, but I want to
publish a lot of new features! :)


Il giorno lun 21 feb 2022 alle ore 14:39 Francesco Sciuti <
***@***.***> ha scritto:

**Comment by NOtherDev** [comment]
Yes @fsciuti, that would most probably work. I did what I came up with back then and used what was at my disposal back then 🙂 I'm happy to deploy Eyedropper myself now and let you prepare the new, easier deployment procedure in a separate PR 🚀 Would this work?

**Comment by fsciuti** [comment]
Great idea! Thank you...First I'll probably work on the new deployment and
then on the migration from Sapper to SvelteKit.

Francesco Sciuti

CEO @ Devmy - Software Factory & Learning Google Developer Expert - Web
Technologies, & Microsoft MVP


[image: Google Developer Expert]
<https://developers.google.com/community/experts/directory/profile/profile-francesco-sciuti>[image:
Microsoft MVP]
<https://mvp.microsoft.com/it-it/PublicProfile/5003974?fullName=Francesco+Sciuti>
<https://www.meetup.com/it-IT/AWS-User-Group-Catania/>


Tel./Whatsapp: +393920445357

*Web: https:// <https://devmy.it/>**devmy.it <https://devmy.it/>*




*Facebook: https://www.facebook.com/Devmy.it
<https://www.facebook.com/Devmy.it>Linkedin:
https://www.linkedin.com/company/devmyfactory/
<https://www.linkedin.com/company/devmyfactory/>Twitter:
https://twitter.com/DevmyFactory
<https://twitter.com/DevmyFactory>Instagram:
https://www.instagram.com/Devmy.it/
<https://www.instagram.com/Devmy.it/>*

*Youtube: https://www.youtube.com/c/devmy <https://www.youtube.com/c/devmy>*



*Francesco è anche:**- Google Certified Certified - Mobile Web Specialist*

*- AWS User Group Leader*


*- Open JS Foundation Supporter*


.    <https://www.meetup.com/it-IT/AWS-User-Group-Catania/>
<https://www.credential.net/e7ba4674-48c3-4442-8af3-449fec881496>


Devmy è un marchio di Acadevmy s.r.l. - P.I. 05537900879

Questa e-mail contiene informazioni di carattere confidenziale rivolte
esclusivamente al destinatario sopra indicato. E' vietato l'uso, la
diffusione, distribuzione o riproduzione da parte di ogni altra persona.
Nel caso aveste ricevuto questa e-mail  per errore, siete pregati di
segnalarlo immediatamente al mittente e distruggere quanto ricevuto senza
farne copia. Qualsivoglia utilizzo non autorizzato del contenuto di questa
e-mail costituisce violazione dell'obbligo di non prendere cognizione della
corrispondenza tra altri soggetti, salvo più grave illecito, ed espone il
responsabile alle relative conseguenze (D. Lgs. 196/2003).

The information contained in this message and any attachments hereto are
intended only for the personal use of the designated recipient(s) named
above. This message is a professional communication, and as such, is
privileged and confidential. If the reader of this message is not the
intended recipient, you are hereby notified that you have received this
document in error, and that any review, dissemination, distribution, or
copying of this message is prohibited. Please notify us immediately by
telephone.


Il giorno gio 24 feb 2022 alle ore 20:45 Adam Bar ***@***.***>
ha scritto:

**Comment by NOtherDev** [comment]
Deployed 💫

**Comment by fsciuti** [comment]
Thank you...I'll try the new deployment on a fork and a github page with no specific domain.
I will be back soon! :D

**Pull request merged**